test/test_torch.py

```diff
@@ -2244,6 +2280,31 @@ def run_test(A_dims, b_dims, cast):
     def test_lu_solve_batched_many_batches(self):
         self._test_lu_solve_batched_many_batches(self, lambda t: t)
 
+    @staticmethod
+    def _test_lu_solve_batched_dims(self, cast, pivot=True):
+        from numpy.linalg import solve
+        from common_utils import random_fullrank_matrix_distinct_singular_value
+
+        def run_test(A_dims, b_dims, cast, pivot):
+            A = random_fullrank_matrix_distinct_singular_value(*A_dims)
+            b = torch.randn(*b_dims)
+            x_exp = torch.as_tensor(solve(A.numpy(), b.numpy()))
+            A, b = cast(A), cast(b)
+            LU_data, LU_pivots = torch.lu(A, pivot=pivot)
+            x = torch.lu_solve(b, LU_data, LU_pivots)
+            self.assertEqual(x, cast(x_exp))
+
+        # test against numpy.linalg.solve
+        run_test((4, 2, 1, 3), (2, 1, 3, 4, 6), cast, pivot)  # no broadcasting
+        run_test((4, 2, 1, 3), (4, 6), cast, pivot)  # broadcasting b
+        run_test((4,), (2, 1, 3, 4, 2), cast, pivot)  # broadcasting A
+        run_test((4, 1, 3, 1), (2, 1, 3, 4, 5), cast, pivot)  # broadcasting A & b
+
+    @skipIfNoLapack
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_lu_solve_batched_dims(self):
```

Comment:
nit: There's already a `test_lu_solve_batched`, maybe call this `test_lu_solve_batched_broadcasting` to make it clearer what this is testing?

Reply:
Sure.